Frends.Community.RabbitMQ/RabbitMQTask.cs

```diff
@@ -86,33 +78,68 @@ public static bool WriteMessage([PropertyTab] WriteInputParams inputParams)
                     basicProperties = _channel.CreateBasicProperties();
                     basicProperties.Persistent = true;
                 }
-                
-                if (inputParams.WriteMessageCount != null &&
-                    _channel.MessageCount(inputParams.QueueName) >= int.Parse(inputParams.WriteMessageCount))
-                {
-                    _channel.TxSelect();
-                    _channel.ConfirmSelect();
-                    _channel.CreateBasicPublishBatch().Publish();
-                    _channel.TxCommit();
-                    if (_channel.MessageCount(inputParams.QueueName) > 0)
-                        _channel.TxRollback();
-                    return true;
-                }
 
-                else
+                
+                if (inputParams.WriteMessageCount == null || 
+                    inputParams.WriteMessageCount != null && int.Parse(inputParams.WriteMessageCount) == 1)
                 {
                     _channel.BasicPublish(exchange:
                         inputParams.ExchangeName,
                         routingKey: inputParams.RoutingKey,
                         basicProperties: basicProperties,
                         body: inputParams.Data);
-                    return false;
+                    return true;
+                }
+                else
+                {
+                    // Add message into a memory based on producer write capability.
+                    if (inputParams.WriteMessageCount != null &&
+                        _channel.MessageCount(inputParams.QueueName) <= int.Parse(inputParams.WriteMessageCount))
+                    {
+                        _channel.CreateBasicPublishBatch().Add(
```

Comment:
Doesnt this create a new batch each time the Task is called, meaning that previous messages are lost?

Reply:
Creates now a batch based on process execution id.